edgecast/resources/cps/certificate.go

```diff
@@ -422,8 +422,53 @@ func ResourceCertificateRead(ctx context.Context,
 	d *schema.ResourceData,
 	m interface{},
 ) diag.Diagnostics {
-	// Not Yet Implemented
 	var diags diag.Diagnostics
+
+	config := m.(**api.ClientConfig)
+	cpsService, err := buildCPSService(**config)
+	if err != nil {
+		return diag.FromErr(err)
+	}
+
+	params := certificate.NewCertificateGetParams()
+	n, err := strconv.ParseInt(d.Get("id").(string), 10, 64)
+	if err == nil {
+		fmt.Printf("%d of type %T", n, n)
+	}
+	params.ID = n
+	resp, err := cpsService.Certificate.CertificateGet(params)
+	if err != nil {
+		d.SetId("")
+		return diag.FromErr(err)
+	}
+
+	log.Printf("[INFO] Retrieved certificate: %# v", pretty.Formatter(resp))
+
+	d.SetId(strconv.FormatInt(resp.ID, 10))
```

Comment:
Do we expect the ID to change after we've used the ID to read the resource? If not, we should remove this I believe to avoid the strange condition where the certificate ID changed unexpectedly.